dataframe/CMakeLists.txt

```
@@ -2,28 +2,32 @@
 file(GLOB sources *.cc)
 file(GLOB headers *.h)
 
-PODIO_GENERATE_DICTIONARY(libedm4hepRDF edm4hepRDF.h SELECTION selection.xml USES edm4hep)
 
-add_library(edm4hepRDF SHARED ${sources} ${headers} libedm4hepRDF.cxx )
-target_link_libraries(edm4hepRDF edm4hep podio::podio ROOT::Physics)
+add_library(edm4hepRDF SHARED ${sources} ${headers})
+target_link_libraries(edm4hepRDF edm4hep ROOT::Physics)
+PODIO_GENERATE_DICTIONARY(edm4hepRDF ${headers} SELECTION selection.xml USES edm4hepRDF
+  OPTIONS --library edm4hepRDF${CMAKE_SHARED_LIBRARY_SUFFIX}
+  )
+set_target_properties(edm4hepRDF-dictgen PROPERTIES EXCLUDE_FROM_ALL TRUE)
+target_sources(edm4hepRDF PRIVATE edm4hepRDF.cxx)
 
 set_target_properties(edm4hepRDF PROPERTIES
   PUBLIC_HEADER "${headers}")
 
 install(TARGETS edm4hepRDF
   # IMPORTANT: Add the library to the "export-set"
   EXPORT edm4hepTargets
-  RUNTIME DESTINATION "${INSTALL_BIN_DIR}" COMPONENT bin
-  LIBRARY DESTINATION "${INSTALL_LIB_DIR}" COMPONENT shlib
-  PUBLIC_HEADER DESTINATION "${INSTALL_INCLUDE_DIR}/edm4hep"
+  RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}" COMPONENT bin
+  LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}" COMPONENT shlib
+  PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/edm4hep"
     COMPONENT dev)
 
 install(FILES
-  "${PROJECT_BINARY_DIR}/dataframe/libedm4hepRDFDict.rootmap"
-  DESTINATION "${INSTALL_LIB_DIR}" COMPONENT dev)
+  "${PROJECT_BINARY_DIR}/dataframe/edm4hepRDFDict.rootmap"
+  DESTINATION "${CMAKE_INSTALL_LIBDIR}" COMPONENT dev)
 
 if (${ROOT_VERSION} GREATER 6)
   install(FILES
       "${PROJECT_BINARY_DIR}/dataframe/libedm4hepRDF_rdict.pcm"
```

```suggestion
      "${PROJECT_BINARY_DIR}/dataframe/edm4hepRDF_rdict.pcm"
```
I'm guessing the `lib` prefix is removed here too?

---

No, the pcm file gets its name from the first argument to PODIO_GENERATE_DICTIONARY, the rootmapfile is set to lib${dictionary}Dict.rootmap. This has to stay to keep avoid to much divergence with the root REFLEX_GENERATE_DICTIONARY

---

I see! But then it seems  we have to change this line here: https://github.com/HSF/EDM4hep/blob/master/edm4hep/CMakeLists.txt#L42

---

Edited, its of course the other way around

---

To be able to merge the dictionary.cxx file into the library we have to tell genreflex that what our library is called, so that name ends up in the rootmap
```
[ libedm4hep.so ]
# List of selected classes
class edm4hep::CaloHitContribution
...
```
Otherwise the GENERATE_DICTIONARY macro uses the libedm4hepDict.so name.
(and changing the macro would diverge from the macro in root, which we maybe can use in the future.)